docs/source/algorithms.rst

```diff
@@ -50,3 +50,85 @@ Using the same example file as before, we can execute a GaussFilter on our test
     gf.filterExperiment(exp)
     # MzMLFile().store("test.filtered.mzML", exp)
 
+
+
+Spectra Merge Algorithm
+*************************
+
+The pyOpenMS spectra merge algorithms allows the merge of several spectra by either increasing S/N ratio (for MS1 and above) or merging scans which stem from similar precursors (for MS2 and above). In any case, the number of scans will be reduced. In addition, pyOpenMS provides the method to average spectra. 
+
+Different spectra merge algorithms are available in pyOpenMS:
+
+- mergeSpectraBlockWise
+- mergeSpectraPrecursors
+- average (over neighbouring spectra for MS1 or above)
+
+In order to have a better picture of the algorithms, each one will be introduced by an example below. We will use another example file compared to above and start with an example of merging spectra blockwise:
+
+.. code-block:: python
+
+    # retrieve data 
+    gh = "https://raw.githubusercontent.com/OpenMS/pyopenms-extra/master"
+    urlretrieve(gh + "/src/data/small.mzML", "test.mzML")
+
+    # load MS data and store as MSExperiment object
+    exp = MSExperiment()
+    MzMLFile().load('test.mzML', exp)
+
+    spectra = exp.getSpectra()
+    print(f"There are {len(spectra)} spectra before merging.\n")
+
+    # merges blocks of MS or MS2 spectra
+    sm = SpectraMerger()
+    sm.mergeSpectraBlockWise(exp)
+
+    spectraMB = exp.getSpectra()
+    print(f"There are {len(spectraMB)} spectra after merging")
+
+
+.. code-block:: output 
+  
+    There are 236 spectra before merging.
+
+    There are 90 spectra after merging. 
+
+
+As predicated, the number of scans decreased after merging MS spectra. The modified data structure can be stored on disk:
+
+.. code-block:: python
+
+    MzMLFile().store("mergedBlockWise.mzML", exp)
+
+
+The merging of spectra with similar precursors progresses likewise:
+
+.. code-block:: python 
+
+    # load MS data and store as MSExperiment object
+    exp = MSExperiment()
+    MzMLFile().load('test.mzML', exp)
+
+    # merge spectra with similar precursors 
+    sm = SpectraMerger()
+    sm.mergeSpectraPrecursors(exp)
+
+    # store modified data 
+    # MzMLFile().store("mergedBlockWise.mzML", exp)
+
+
+``SpectraMerger`` presents a method to average experimental data over neighbouring spectra. The averaging types to be used by the method are either ``gaussian`` or ``tophat``:
+
+.. code-block:: python 
+
+    # load MS data and store as MSExperiment object
+    exp = MSExperiment()
+    MzMLFile().load('test.mzML', exp)
+
+    # average spectra with gaussian
+    sm = SpectraMerger()
+    sm.average(exp, "gaussian")  
```

Comment:
What is the default number of spectra being merged?
Why no before/after comparison here?

Reply:
There were no changes in the spectra numbers with the example file.

Reply:
haha ok. Doesn't that sound like something is wrong? Please check if this does what it is supposed to do.

Reply:
Yes, as @timosachsenberg suggested, the outcome of the average depends on the MS level. There is no change in the spectra number at MS2 level, whereas the total number of spectra changes to 183 for the MS1 level.

Reply:
I am just looking for simple explanations: e.g.
Per default, SpectraMerger merges MS1. Before we had 6 MS1 spectra. Now we have 2 MS1 spectra, because per default SpectraMerger always merges 3 consecutive spectra.
Or:
The number of resulting spectra cannot be predicted since the algorithm combines spectra based on property XYZ.

Reply:
Thank you for the feedback! Then I will take this into account.

Reply:
I did not check how complex the code is but if feasible you can take a look and find out what the differences are between the different modes and extend the documentation in the header file.

Reply:
The code is not very complex, but it takes a bit of time due to the different functions, but it's still manageable. I am also currently reading the code to understand the differences between each method better. Yes, I can do that.